applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_adjoint_solver.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as KCPFApp
+from KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_solver import PotentialFlowSolver
+import KratosMultiphysics.StructuralMechanicsApplication
+
+def CreateSolver(model, custom_settings):
+    return PotentialFlowAdjointSolver(model, custom_settings)
+
+class PotentialFlowAdjointSolver(PotentialFlowSolver):
+    def __init__(self, model, custom_settings):
+
+        self.response_function_settings = custom_settings["response_function_settings"].Clone()
+        self.sensitivity_settings = custom_settings["sensitivity_settings"].Clone()
+        custom_settings.RemoveValue("response_function_settings")
+        custom_settings.RemoveValue("sensitivity_settings")
```

Comment:
we usually try to avoid using `RemoveValue` and `Clone`
the reason is that you will not be able to print the validated configuration after constructing the solver

Reply:
Right now it is done like this so that I can later call the parent solver to initialize it, so response_function and sensitivity_settings are removed to avoid errors. (in fact this is what it's done in the structural_mechanics_adjoint_solver).

What would you suggest so that you can call the parent's intializer without getting errors? Copying again the settings after removing them? 

In any case, maybe it would be good to keep the same functionality/workflow as the current structural adjoint solver, and discuss this kind of improvements in separate PRs altogether (modifying both this solver and the structural one).  Do you agree?

Reply:
I see. Also I agree abt having the same workflow as in the structural. Just saying that is should be avoided, I know that this is not always possible

I am not proposing a solution as I don't have one :)

Maybe this can be improved in a future PR, also ping @armingeiser @MFusseder

Reply:
I am aware of this, but i also don't have a nice solution. I would also keep it like it is for now, for the reasons you mentioned.